src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCache.cs

```diff
@@ -178,83 +220,158 @@ public static IEnumerable<T> EnumerateAllComponentsInScenesAndPrefabs<T>()
         }
     }
 
-    internal abstract class AssetCache<TAssetEntry, TAsset> : AssetCache, IAssetCache
-        where TAssetEntry : AssetCacheEntry<TAsset>, new()
+    internal abstract class AssetCache<TAsset> : AssetCache, IAssetCache
         where TAsset : UnityEngine.Object
     {
-        [SerializeField]
-        private TAssetEntry[] assets = null;
+        const string assetContentExtension = ".asset";
 
-        private Dictionary<AssetId, TAssetEntry> lookupByAssetId;
-        private Dictionary<TAsset, TAssetEntry> lookupByAsset;
+#pragma warning disable 414
+        [SerializeField]
+        private NameEntry[] assets = null;
+#pragma warning restore 414
 
-        protected IDictionary<AssetId, TAssetEntry> LookupByAssetId
+        private Dictionary<string, List<AssetId>> LookupByName

```

Comment:
>LookupByName [](start = 50, length = 12)

Just to check (haven't read through all the code yet, so I'm leaving this comment with the knowledge it might get resolved after reading,  lookupByName gets populated later down the line correctly as "assets" get filled right? i.e. there's no danger of accessing LookupByName early (before assets is non-null?)

Reply:
I'll add some error logging here. assets being null can't be fixed at runtime. These AssetCaches are stored in the Resources folder so assets should already be populated when this AssetCache is created.

---
In reply to: [355038927](https://github.com/microsoft/MixedReality-SpectatorView/pull/295#discussion_r355038927) [](ancestors = 355038927)